app/controllers/laboratories_controller.rb

```diff
@@ -97,4 +55,25 @@ def destroy
       render status: 500
     end
   end
+
+  private
+
+  def check_role
+    redirect_to(root_url) && return unless current_user.can_edit_patient_laboratories?
+  end
+
+  def check_patient
+    patient_id = params.permit(:patient_id)[:patient_id]&.to_i
```

Comment:
if you make this `@patient_id` you can use it in all the above methods and remove this line in all of them: `patient_id = params.permit(:patient_id)[:patient_id]&.to_i`